kbase/narrative
# DATAUP-197 app cell cancel after code compile error

# Description of PR purpose/changes

If a user makes a change to the code in an app cell, and the code does not compile in the app cell, the user can now press cancel. Users were not able to do this before. 

I also added a could of tests to kick off testing for the app cell. This is not comprehensive, and should be added on over time. 

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-197

# Testing Instructions
Execute `make test-frontend-unit`

To test the changes through the ui complete the following steps:
1. open a narrative
2. add an app cell instance to the narrative
3. Open the app cell code, and make a change that will cause the code to fail to compile. 
4. Click run
5. Click cancel when you see the error
6. You can now rerun the cell


# Dev Checklist:


# Updating Version and Release Notes (if applicable)
